crates/modelardb_compression/src/models/gorilla.rs

```diff
@@ -128,15 +134,20 @@ impl Gorilla {
                 );
             } else {
                 // Store the leading zero bits before the meaningful bits using
-                // 5 and 6 bits respectively as described in the Gorilla paper.
+                // 4 and 5 bits respectively as implemented in the ELF repository.
                 self.compressed_values.append_a_one_bit();
                 self.compressed_values
-                    .append_bits(leading_zero_bits as u64, 5);
+                    .append_bits(leading_zero_bits as u64, 4);
 
                 let meaningful_bits =
                     models::VALUE_SIZE_IN_BITS - leading_zero_bits - trailing_zero_bits;
-                self.compressed_values
-                    .append_bits(meaningful_bits as u64, 6);
+                // 0 is used to denote 32 meaningful bits.
+                if meaningful_bits == 32 {
```

Comment:
Is a branch needed? Could it not be computed as `meaningful_bits % 32` to avoid the branch?